tests/unit/features/partitions/test_lifecycle_manager.py

```diff
@@ -154,7 +197,42 @@ def test_partitions_invalid(self, new_dir, parts_data, partitions):
         """Raise an error if partitions are not lowercase alphabetical characters."""
         with pytest.raises(
             errors.FeatureError,
-            match=r"Partition '[\w-]*' must only contain lowercase letters.*",
+            match=(
+                r"Partition '[\w-]*' is invalid.\n"
+                r"Partitions must only contain lowercase letters.*"
+            ),
+        ):
+            lifecycle_manager.LifecycleManager(
+                parts_data,
+                application_name="test_manager",
+                cache_dir=new_dir,
+                partitions=partitions,
+            )
+
+    @pytest.mark.parametrize(
+        "partitions",
+        [
+            ["default", "/"],
+            ["default", "/a"],
+            ["default", "a/"],
+            ["default", "a/b/"],
+            ["default", "/a/b"],
+            ["default", "a//b"],
+            ["default", "/a/"],
+            ["default", "a/Test"],
+            ["default", "a/TEST"],
+            ["default", "a/test1"],
+            ["default", "Test/a"],
+            ["default", "TEST/a"],
+            ["default", "test1/a"],
+            ["default", "te-st/a"],
```

Comment:
Shouldn't we also test for namespaced partition names starting or ending with dash such as `a/-b`, `a/b-`, or even just `-`, `-/b`, etc?

Reply:
Good idea.  I added many more invalid test cases with hyphens.